pkg/eigenState/operatorAVSSplits/operatorAVSSplits.go

```diff
@@ -0,0 +1,279 @@
+	return nil
+}
+
+func (oas *OperatorAVSSplitModel) CleanupProcessedStateForBlock(blockNumber uint64) error {
+	delete(oas.stateAccumulator, blockNumber)
+	return nil
+}
+
+func (oas *OperatorAVSSplitModel) HandleStateChange(log *storage.TransactionLog) (interface{}, error) {
+	stateChanges, sortedBlockNumbers := oas.GetStateTransitions()
+
+	for _, blockNumber := range sortedBlockNumbers {
+		if log.BlockNumber >= blockNumber {
+			oas.logger.Sugar().Debugw("Handling state change", zap.Uint64("blockNumber", log.BlockNumber))
+
+			change, err := stateChanges[blockNumber](log)
+			if err != nil {
+				return nil, err
+			}
+			if change == nil {
+				return nil, nil
+			}
+			return change, nil
+		}
+	}
+	return nil, nil
+}
+
+// prepareState prepares the state for commit by adding the new state to the existing state.
+func (oas *OperatorAVSSplitModel) prepareState(blockNumber uint64) ([]*OperatorAVSSplit, error) {
+	accumulatedState, ok := oas.stateAccumulator[blockNumber]
+	if !ok {
+		err := xerrors.Errorf("No accumulated state found for block %d", blockNumber)
+		oas.logger.Sugar().Errorw(err.Error(), zap.Error(err), zap.Uint64("blockNumber", blockNumber))
+		return nil, err
+	}
+
+	recordsToInsert := make([]*OperatorAVSSplit, 0)
+	for _, split := range accumulatedState {
+		recordsToInsert = append(recordsToInsert, split)
+	}
+	return recordsToInsert, nil
+}
+
+// CommitFinalState commits the final state for the given block number.
+func (oas *OperatorAVSSplitModel) CommitFinalState(blockNumber uint64) error {
+	recordsToInsert, err := oas.prepareState(blockNumber)
+	if err != nil {
+		return err
+	}
+
+	if len(recordsToInsert) > 0 {
+		for _, record := range recordsToInsert {
+			res := oas.DB.Model(&OperatorAVSSplit{}).Clauses(clause.Returning{}).Create(&record)
+			if res.Error != nil {
+				oas.logger.Sugar().Errorw("Failed to insert records", zap.Error(res.Error))
+				return res.Error
+			}
+		}
+	}
+	return nil
+}
+
+// GenerateStateRoot generates the state root for the given block number using the results of the state changes.
+func (oas *OperatorAVSSplitModel) GenerateStateRoot(blockNumber uint64) (types.StateRoot, error) {
+	inserts, err := oas.prepareState(blockNumber)
+	if err != nil {
+		return "", err
+	}
+
+	inputs := oas.sortValuesForMerkleTree(inserts)
+
+	if len(inputs) == 0 {
+		return "", nil
+	}
+
+	fullTree, err := oas.MerkleizeState(blockNumber, inputs)
+	if err != nil {
+		oas.logger.Sugar().Errorw("Failed to create merkle tree",
+			zap.Error(err),
+			zap.Uint64("blockNumber", blockNumber),
+			zap.Any("inputs", inputs),
+		)
+		return "", err
+	}
+	return types.StateRoot(utils.ConvertBytesToString(fullTree.Root())), nil
+}
+
+func (oas *OperatorAVSSplitModel) sortValuesForMerkleTree(splits []*OperatorAVSSplit) []*base.MerkleTreeInput {
+	inputs := make([]*base.MerkleTreeInput, 0)
+	for _, split := range splits {
+		slotID := base.NewSlotID(split.TransactionHash, split.LogIndex)
+		value := "added"
```

Comment:
rather than "added" let's use `split.NewOperatorAVSSplitBips` to capture the actual value of the change

Reply:
What's the purpose of the `value` field here?

Reply:
The value is what is merkelized as a leaf of the tree. So right now you end up with `<slotId>:added` as your leaf which isnt all that descriptive or meaningful. `<slotId>:<new bips value>` is much more meaningful and captures more of the actual change. This way if we ever needed to debug an issue, you could in theory know if the bips value that was added is wrong or not.

Reply:
Done in [997e1b3](https://github.com/Layr-Labs/sidecar/pull/112/commits/997e1b36c45bcf628de0e2e26696f535d57fff24)